manifests/params.pp

```diff
@@ -195,10 +195,21 @@
 
   # Puppet service name
   $service_name = 'puppet'
-  $agent_restart_command = $::osfamily ? {
-    'Debian' => '/usr/sbin/service puppet reload',
-    'Redhat' => '/usr/sbin/service puppet reload',
-    default  => undef,
+  # Command to restart the agent
+  case $::osfamily {
+    'Debian' : {
+      $agent_restart_command = "/usr/sbin/service ${service_name} reload"
+    }
+    'Redhat' : {
+      $osreleasemajor = regsubst($::operatingsystemrelease, '^(\d+)\..*$', '\1') # workaround for the possibly missing operatingsystemmajrelease
+      $agent_restart_command = $osreleasemajor ? {
+        '7'     => "/usr/bin/systemctl restart ${service_name}",
```

Comment:
The original change was to use the "softer" reload in order to avoid killing the agent while configuring puppet itself.